pegen/parse_string.c

```diff
@@ -472,13 +486,27 @@ fstring_fix_expr_location(Token *parent, expr_ty n, char *expr_str)
         }
         substr = strstr(parent_str, expr_str);
         if (substr) {
-            start = substr;
-            while (start > parent_str) {
-                if (start[0] == '\n')
+            // The following is needed, in order to correctly shift the column
+            // offset, in the case that (disregarding any whitespace) a newline
+            // immediately follows the opening curly brace of the fstring expression.
+            int newline_after_brace = 1;
+            start = substr + 1;
+            while (start && *start != '}' && *start != '\n') {
+                if (!(*start == ' ' || *start == '\t' || *start == '\f')) {
```

Comment:
Suggestion, introduce the "!" inside the condition to simplify the logic (so it reads directly left to right without the need to negate everything at the end) :

```
(*start != ' ' && *start != '\t' && *start != '\f')
```